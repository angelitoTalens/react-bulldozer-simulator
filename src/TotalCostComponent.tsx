import React from "react";
import "./styles/TotalCostComponent.scss"
import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";
import { Cost } from "./EnumTypes";


type TotalCostComponentProps = {
    totalFuel: Cost;
    communication: Cost;
    unclearedBlock: Cost;
    clearProtectedTree: Cost;
    paintDamage: Cost;
}

function TotalCostComponent(props: TotalCostComponentProps) {

    const totalCost = props.totalFuel.value +
                      props.communication.value +
                      props.unclearedBlock.value +
                      props.clearProtectedTree.value +
                      props.paintDamage.value;

    return (
        <Container className="total-cost-container">
            <Table size="sm" striped hover>
                <thead>
                    <tr>
                        <th style={{width: "40%"}}>Item</th>
                        <th style={{width: "30%"}}>Quantity</th>
                        <th style={{width: "30%"}}>Cost (Credit(s))</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>Communication Overhead</td>
                        <td>{props.communication.quantity}</td>
                        <td>{props.communication.value}</td>
                    </tr>
                    <tr>
                        <td>Fuel Usage</td>
                        <td>{props.totalFuel.quantity}</td>
                        <td>{props.totalFuel.value}</td>
                    </tr>
                    <tr>
                        <td>Uncleared Squares</td>
                        <td>{props.unclearedBlock.quantity}</td>
                        <td>{props.unclearedBlock.value}</td>
                    </tr>
                    <tr>
                        <td>Cleared Preserved Tree(s)</td>
                        <td>{props.clearProtectedTree.quantity}</td>
                        <td>{props.clearProtectedTree.value}</td>
                    </tr>
                    <tr>
                        <td>Paint Damage Cost</td>
                        <td>{props.paintDamage.quantity}</td>
                        <td>{props.paintDamage.value}</td>
                    </tr>
                    <tr>
                        <td><b>Total Cost</b></td>
                        <td></td>
                        <td><b>{totalCost}</b></td>
                    </tr>
                </tbody>
            </Table>
        </Container>
    );
}

export default TotalCostComponent;