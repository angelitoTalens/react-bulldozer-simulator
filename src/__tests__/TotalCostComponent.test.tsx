import React from 'react';
import renderer from 'react-test-renderer';

import TotalCostComponent from '../TotalCostComponent';

test ("Test TotalCostComponent", ()  => {

    const component = renderer.create(
        <TotalCostComponent 
            totalFuel={{quantity:4, value:40}}
            communication={{quantity:4, value:40}}
            unclearedBlock={{quantity:1, value:10}}
            clearProtectedTree={{quantity:0, value:0}}
            paintDamage={{quantity:2, value:20}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

