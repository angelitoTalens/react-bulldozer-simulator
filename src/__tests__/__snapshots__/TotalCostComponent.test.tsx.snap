// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TotalCostComponent 1`] = `
<div
  className="total-cost-container container"
>
  <table
    className="table table-sm table-striped table-hover"
  >
    <thead>
      <tr>
        <th
          style={
            Object {
              "width": "40%",
            }
          }
        >
          Item
        </th>
        <th
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Quantity
        </th>
        <th
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Cost (Credit(s))
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Communication Overhead
        </td>
        <td>
          4
        </td>
        <td>
          40
        </td>
      </tr>
      <tr>
        <td>
          Fuel Usage
        </td>
        <td>
          4
        </td>
        <td>
          40
        </td>
      </tr>
      <tr>
        <td>
          Uncleared Squares
        </td>
        <td>
          1
        </td>
        <td>
          10
        </td>
      </tr>
      <tr>
        <td>
          Cleared Preserved Tree(s)
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Paint Damage Cost
        </td>
        <td>
          2
        </td>
        <td>
          20
        </td>
      </tr>
      <tr>
        <td>
          <b>
            Total Cost
          </b>
        </td>
        <td />
        <td>
          <b>
            110
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
